# DSB Repair Kinetics Calculator

The official R scripts for analysing designer nuclease-induced DSB repair kinetics data generated by [MEGA safety profiling](https://github.com/A-Chalk/MEGA). Novel programmable endonucleases are rapidly being produced to facilitate gene therapy development. Studying the kinetics of their DNA cleavage activity and subsequent cell repair can elucidate essential safety and efficacy information.

## Publications

- [MEGA Methodology and Technique Insights (TBD)](https://www.nature.com)
- [Protocol Paper (TBD)](https://www.nature.com)

## File descriptions

* **$k$ Calc**: Solves for $k$ values based on observed data and fits curves (_individual sample_)
* **$k$ Calc+**: Utilise cell repair inhibitors to improve $k_{dsb}$ and $k_{pr}$ estimations (_RNP Only_ & _RNP + Inhibitor_ samples)
* **DSB Kinetics Visualiser**: Excel template that visualises and quantifies fitted kinetics curves
* **Allele Calculator**: Plug in any $k$ values, useful for studying model dynamics

## DSB Calculator Applications
* Measure precise DNA repair and recurrent Cas9 cleavage
* Continuously estimate the rates, probabilities, and allele frequencies of any DSB repair products
* Optimize the speed and safety of novel gene editing platforms
* Elucidate deeper biological mechanisms of DSB repair

## Schematic

<img width="600" align="center" alt="MEGA Overview" src="https://github.com/A-Chalk/DSB-Kinetics-Calculator/assets/167306438/c1fd58b2-a67c-4b29-aa92-d06c2bfdc6e4">

**a.** Programmable endonucleases recognise and cleave target sequences, producing a number of intended and unintended DSB repair products. **b.** If the cell manages to repair an induced DSB, it can either generate a reaction terminating mutation (no longer recognized by the endonuclease), or it can precisely restore the wildtype sequence, which can then be reidentified and cleaved again. This recurrent cleavage is what ultimately drives the generation of mutation products. **c.** By sampling from an edited cell population over a timeseries and processing the DNA with MEGA safety screening, a model can be generated describing the activity of DSB generation and cellular repair.

## Model Description

This pipeline employs the three-state model of DSB repair in which Cas9 cleavage, precise DNA repair, and mutatagenic repair are each constrained by fixed rate coeffecients $k_{dsb}(t)$, $k_{pr}$, and $k_{m}$ respectively.

$$Wildtype \xrightleftharpoons[k_{pr}]{k_{dsb}(t)}DSB\xrightarrow{k_m}Mutation$$

There are many possible reaction terminating products ($k_m$) detected by MEGA, including indels, large deletions, and targeted gene integration, each with distinct rate coeffecients $k_{in}$, $k_{ld}$, $k_{ti}$. 

This model can be described continuously across the timeseries by a set of ordinary differential equations:

* Wildtype rate | $\frac{dWildtype}{dt} = (k_{pr} * DSB)-(k_{dsb} * D(t) * WT)$<br/>
* DSB rate | $\frac{dDSB}{dt} = (k_{dsb} * D(t) * WT) - (k_{pr} + k_{in} + k_{ld}) * DSB$<br/>
* Precise repair rate | $\frac{dPR}{dt} = (k_{pr} * DSB)-(k_{dsb} * D(t) * PR)$<br/>
* Indel rate | $\frac{dIndel}{dt} = k_{in} * DSB$<br/>
* Large deletion rate | $\frac{dLD}{dt} = k_{ld} * DSB$<br/>
* Targeted integration rate | $\frac{dTI}{dt} = k_{ti} * DSB$<br/>
* Cas9 trafficking delay | $D(t) = 1 - 2^{-(\frac{t}{delay})}$

The pipeline uses these ODEs to estimate the $k$ values based on an your observed allele population frequencies. Notice that $k_{dsb}$ is multiplied by a delay function $D(t)$ which takes into account a brief initial lag in Cas9 cleavage activity resulting from RNP trafficking into the nucleus post-electroporation (this variable can adjust for slower/faster Cas9 delivery strategies).

## Experimental Setup

Gene editing strategy and timeseries design
* Cell culturing, RNP complexing, inhibitors

The most accurate kinetics information can be generated by editing your cell population alongside repair inhibitors that block cell repair. Our approach involves nucleofecting cells with complexed RNP, splitting the cells, and adding AZD7648 ART558 in DMSO to half. Additionally, if you are interested in studying donor template integration, you can split each group again and transfect half with an AAV (etc). 

This will produce **five** populations to track over your timecourse: Mock untreated cells, RNP-only, RNP + Inhibitors, RNP + DNA donor, and RNP + Inhibitors + DNA donor. 


**MEGA profiling**
Any relevant information regarding plate setup and gathering data from dPCR
(color number, mock, reference, etc)

## Analysis Pipeline
* Data processing
* Data analysis

Example dataset and analysis spreadsheet will be included as downloadable .zip files

## Authors
* J. Alexander Chalk
* Nathan White
* Giandomenico Turchiano

## License

## References
1. Brinkman, E. K. et al. Kinetics and Fidelity of the Repair of Cas9-Induced Double-Strand DNA Breaks. Mol. Cell 70, 801-813.e6 (2018).
2. Ben-Tov, D. et al. Uncovering the Dynamics of Precise Repair at CRISPR/Cas9-induced Double-Strand Breaks. bioRxiv 2023.01.10.523377 (2023)
3. Cucinotta, F. A., Pluth, J. M., Anderson, J. A., Harper, J. V. & O’Neill, P. Biochemical Kinetics Model of DSB Repair and Induction of γ-H2AX Foci by Non-homologous End Joining. Radiat. Res. 169, 214–222 (2008).
4. Rose, J. C. et al. Rapidly inducible Cas9 and DSB-ddPCR to probe editing kinetics. Nat. Methods 14, 891–896 (2017).

